provider-chat/src/java/org/sakaiproject/archiver/provider/ChatArchiver.java

```diff
@@ -0,0 +1,157 @@
+public class ChatArchiver implements Archiveable {
+
+	private static final String CHAT_TOOL = "sakai.chat";
+
+	public void init() {
+		ArchiverRegistry.getInstance().register(CHAT_TOOL, this);
+	}
+
+	public void destroy() {
+		ArchiverRegistry.getInstance().unregister(CHAT_TOOL);
+	}
+
+	@Setter
+	private ChatManager chatManager;
+
+	@Setter
+	private ArchiverService archiverService;
+
+	@Setter
+	private UserDirectoryService userDirectoryService;
+
+	@Override
+	public void archive(final String archiveId, final String siteId, final String toolId, final boolean includeStudentContent) {
+
+		List<ChatChannel> chatChannels = chatManager.getContextChannels(siteId, true);
+
+		for (ChatChannel chatChannel : chatChannels) {
+
+			// Save the metadata for this channel
+			ChatChannelMetadata metadata = new ChatChannelMetadata();
+			metadata.setDateCreated(chatChannel.getCreationDate());
+			metadata.setStartDate(chatChannel.getStartDate());
+			metadata.setEndDate(chatChannel.getEndDate());
+			metadata.setTitle(chatChannel.getTitle());
+			metadata.setDescription(chatChannel.getDescription());
+			metadata.setId(chatChannel.getId());
+			this.archiverService.archiveContent(archiveId, siteId, toolId, Jsonifier.toJson(metadata).getBytes(), 
+					chatChannel.getTitle() + " (metadata).json");
+
+			int numMessages = chatManager.getChannelMessagesCount(chatChannel, null, null);
+
+			// Go through and get chat messages, 99 at a time (i.e. 0-99, 100-199, etc)
+			for (int start = 0; start <= numMessages - (numMessages%100); start += 100) {
+
+				try {
+					List<ChatMessage> chatMessages = chatManager.getChannelMessages(chatChannel, null, null, start, 99, true);
+
+					List<SimpleChatMessage> messagesToSave = createArchiveItem(chatMessages);
+					
+					// Convert to JSON and save to file
+					int rangeStart = start+1;
+					int rangeEnd = numMessages - start >= 100 ? start+100 : numMessages+1;
+					this.archiverService.archiveContent(archiveId, siteId, toolId, Jsonifier.toJson(messagesToSave).getBytes(), 
+							chatChannel.getTitle() + "(" + rangeStart + "-" + rangeEnd + ").json", "Chat messages for " + chatChannel.getTitle());
+
+				} catch (PermissionException e) {
+					log.error("Could not retrieve some chat messages for channel: " + chatChannel.getTitle());
+					continue;
+				}
+			}
+		}
+	}
+	
+	/**
+	 * Build the list of messages to be saved for this channel
+	 * 
+	 * @param chatMessages
+	 * @return the list of messages to be saved
+	 */
+	private List<SimpleChatMessage> createArchiveItem(List<ChatMessage> chatMessages) {
+		List<SimpleChatMessage> messagesToSave = new ArrayList<SimpleChatMessage>();
+		for (ChatMessage message : chatMessages) {
+			SimpleChatMessage simpleChatMessage = new SimpleChatMessage();
+			simpleChatMessage.setBody(message.getBody());
+			simpleChatMessage.setDate(message.getMessageDate());
+			try {
+				simpleChatMessage.setOwner(userDirectoryService.getUser(message.getOwner()).getDisplayName());
```

Comment:
If you have a list, call it something plural. Then inside you make items.those can be a method on their own that return just one thing.

So createArchiveItems and createArchiveItem